animal-sniffer/src/main/java/org/codehaus/mojo/animal_sniffer/SignatureChecker.java

```diff
@@ -620,10 +620,11 @@ private boolean find( Clazz c , String sig , boolean baseFind  )
                                 + ": Covariant return type change detected: "
                                 + toSourceForm( c.getName(), oldSignature ) + " has been changed to "
                                 + toSourceForm( c.getName(), sig ) );
-                            return true;
+                            return false;
                         }
                     }
                 }
+                return true;
```

Comment:
this should still return false

you can remove both return statements and let it fall through to return false (while keeping the logging)